Mathlib/SetTheory/Ordinal/Nimber.lean

```diff
@@ -0,0 +1,359 @@
+- Show the nimbers are algebraically closed.
+-/
+
+universe u v
+
+open Function Order
+
+noncomputable section
+
+/-! ### Basic casts between `Ordinal` and `Nimber` -/
+
+/-- A type synonym for ordinals with natural addition and multiplication. -/
+def Nimber : Type _ :=
+  Ordinal deriving Zero, Inhabited, One, WellFoundedRelation
+
+instance Nimber.linearOrder : LinearOrder Nimber := {Ordinal.linearOrder with}
+instance Nimber.succOrder : SuccOrder Nimber := {Ordinal.instSuccOrder with}
+instance Nimber.orderBot : OrderBot Nimber := {Ordinal.orderBot with}
+instance Nimber.noMaxOrder : NoMaxOrder Nimber := {Ordinal.noMaxOrder with}
+instance Nimber.zeroLEOneClass : ZeroLEOneClass Nimber := {Ordinal.zeroLEOneClass with}
+
+/-- The identity function between `Ordinal` and `Nimber`. -/
+@[match_pattern]
+def Ordinal.toNimber : Ordinal ≃o Nimber :=
+  OrderIso.refl _
+
+/-- The identity function between `Nimber` and `Ordinal`. -/
+@[match_pattern]
+def Nimber.toOrdinal : Nimber ≃o Ordinal :=
+  OrderIso.refl _
+
+namespace Nimber
+
+open Ordinal
+
+@[simp]
+theorem toOrdinal_symm_eq : Nimber.toOrdinal.symm = Ordinal.toNimber :=
+  rfl
+
+@[simp]
+theorem toOrdinal_toNimber (a : Nimber) :
+    Ordinal.toNimber (Nimber.toOrdinal a) = a := rfl
+
+theorem lt_wf : @WellFounded Nimber (· < ·) :=
+  Ordinal.lt_wf
+
+instance : WellFoundedLT Nimber :=
+  Ordinal.wellFoundedLT
+
+instance : IsWellOrder Nimber (· < ·) :=
+  { }
+
+instance : ConditionallyCompleteLinearOrderBot Nimber :=
+  WellFoundedLT.conditionallyCompleteLinearOrderBot _
+
+@[simp]
+theorem bot_eq_zero : ⊥ = 0 :=
+  rfl
+
+@[simp]
+theorem toOrdinal_zero : toOrdinal 0 = 0 :=
+  rfl
+
+@[simp]
+theorem toOrdinal_one : toOrdinal 1 = 1 :=
+  rfl
+
+@[simp]
+theorem toOrdinal_eq_zero (a) : toOrdinal a = 0 ↔ a = 0 :=
+  Iff.rfl
+
+@[simp]
+theorem toOrdinal_eq_one (a) : toOrdinal a = 1 ↔ a = 1 :=
+  Iff.rfl
+
+@[simp]
+theorem toOrdinal_max {a b : Nimber} : toOrdinal (max a b) = max (toOrdinal a) (toOrdinal b) :=
+  rfl
+
+@[simp]
+theorem toOrdinal_min {a b : Nimber} : toOrdinal (min a b) = min (toOrdinal a) (toOrdinal b) :=
+  rfl
+
+theorem succ_def (a : Nimber) : succ a = toNimber (toOrdinal a + 1) :=
+  rfl
+
+/-- A recursor for `Nimber`. Use as `induction x`. -/
+@[elab_as_elim, cases_eliminator, induction_eliminator]
+protected def rec {β : Nimber → Sort*} (h : ∀ a, β (toNimber a)) : ∀ a, β a := fun a =>
+  h (toOrdinal a)
+
+/-- `Ordinal.induction` but for `Nimber`. -/
+theorem induction {p : Nimber → Prop} : ∀ (i) (_ : ∀ j, (∀ k, k < j → p k) → p j), p i :=
+  Ordinal.induction
+
+protected theorem le_zero {a : Nimber} : a ≤ 0 ↔ a = 0 :=
+  Ordinal.le_zero
+
+protected theorem not_lt_zero (a : Nimber) : ¬ a < 0 :=
+  Ordinal.not_lt_zero a
+
+protected theorem pos_iff_ne_zero {a : Nimber} : 0 < a ↔ a ≠ 0 :=
+  Ordinal.pos_iff_ne_zero
+
+instance (a : Nimber.{u}) : Small.{u} (Set.Iio a) :=
+  Ordinal.small_Iio a
+
+theorem not_small_nimber : ¬ Small.{u} Nimber.{u} :=
```

Comment:
```suggestion
theorem not_small_nimber : ¬ Small.{u} Nimber.{max u v} :=
```
This is `Nimber.not_small_nimber` compared to `_root_.not_small_ordinal`, I think this should be outside the namespace.